util.go

```diff
@@ -24,6 +24,10 @@ const (
 	SnowflakeRequestIDKey contextKey = "SNOWFLAKE_REQUEST_ID"
 	// streamChunkDownload determines whether to use a stream based chunk downloader
 	streamChunkDownload paramKey = "STREAM_CHUNK_DOWNLOAD"
+	//FetchResultByID the queryID of query result to fetch
+	FetchResultByID paramKey = "SF_FETCH_RESULT_BY_ID"
```

Comment:
the query id is always a UUID and has to be a properly formatted UUID (cus on GS side we generate it as UUID)
I noticed that we store this as a `str` in the context value, can we store it as a `UUID` so that we dont even need to do the regexp validation?